model-optimizer/mo/middle/passes/mean_scale_values.py

```diff
@@ -20,62 +20,130 @@
+    # # Input->data->Mul->scsh_data    =>    Input->scsh_data
+    # # Input->data->Add->scsh_data    =>    Input->scsh_data
     input_op = match['input_op']
-    scale_shift = match['scale_shift']
-    weights = np.squeeze(match['weights'].value)
-    biases = np.squeeze(match['biases'].value)
 
-    if graph.graph['cmd_params'].reverse_input_channels:
-        biases = np.flip(biases)
-
-    if any([x != 1 for x in weights]):
-        return
-
-    # Keep biases (mean values) for current input as graph attr and remove ScaleShift layer
-    # Input->data->ScaleShift->scsh_data    =>    Input->scsh_data
-    graph.remove_edge(input_op.id, input_op.out_node().id)
-    graph.add_edge(input_op.id, scale_shift.out_node().id, out=0)
-    graph.remove_edge(scale_shift.id, scale_shift.out_node().id)
-
-    # If bias contains zeros we just remove it
-    if all([x == 0 for x in biases]):
-        return
+    if 'add' in match:
+        add_outputs = list(match['add'].out_ports().values())
+        assert len(add_outputs) > 0
+        match['add'].out_port(0).get_connection().set_source(input_op.out_port(0))
+        if 'mul' in match:
+            graph.remove_node(match['mul'].id)
+        else:
+            graph.remove_node( match['add'].id)
+    elif 'mul' in match:
+        mul_outputs = list(match['mul'].out_ports().values())
+        assert len(mul_outputs) > 0
+        match['mul'].out_port(0).get_connection().set_source(input_op.out_port(0))
+        graph.remove_node(match['mul'].id)
+
+    if 'const_add_output' in match and match['const_add_output'].has('value'):
+        biases = match['const_add_output'].value
+
+        if graph.graph['cmd_params'].reverse_input_channels:
+            biases = np.flip(biases)
+
+        # If bias contains zeros we just remove it
+        if all([x == 0 for x in biases]):
+            return
+
+        biases *= -1
+        # In pre-process section, mean_values are subtracted
+        mean_values.update({input_op.name: np.array(biases)})
+
+        # Add graph attribute 'mean_values' that stores mean_values per input if exists
+        if graph.graph.get('mean_values', None):
+            graph.graph['mean_values'].update(mean_values)
+        else:
+            graph.graph['mean_values'] = mean_values
+
+
+def move_mean_values_to_preprocess(graph: Graph):
+    """
+    This function finds mean values layers after input layer and if it has weights with ones, it deletes mean values layers
+    and creates graph dict attribute : {'input':np.array(...), 'input2': ... }
+    """
+    apply_pattern(
+        graph,
+        nodes = [
+            ('input_op', dict(kind='op', op='Parameter')),
+            ('input_output', dict(kind='data')),
 
-    # In pre-process section, mean_values are subtracted
-    biases *= -1
+            ('const_mul', dict(kind='op', op='Const')),
+            ('const_mul_output', dict(kind='data')),
+            ('mul', dict(kind='op', op='Mul')),
```

Comment:
I just realised that this pattern with `input->Mul` is useless because there is nothing to move to pre-process because there are no mean values. Please, remove it.